src/core/libmaven/jsonReports.cpp

```diff
@@ -1,286 +1,678 @@
+            myfile << ",\n" << "\"noNoiseObs\": " << peak->noNoiseObs ;
+            myfile << ",\n" << "\"signalBaselineRatio\": " << peak->signalBaselineRatio ;
+            myfile << ",\n" << "\"fromBlankSample\": " << peak->fromBlankSample ;
+            myfile << ",\n" << "\"peakAreaFractional\": " << peak->peakAreaFractional ;
+            myfile << ",\n" << "\"symmetry\": " << peak->symmetry ;
+            myfile << ",\n" << "\"noNoiseFraction\": " << peak->noNoiseFraction ;
+            myfile << ",\n" << "\"groupOverlap\": " << peak->groupOverlap ;
+            myfile << ",\n" << "\"groupOverlapFrac\": " << peak->groupOverlapFrac ;
+            myfile << ",\n" << "\"gaussFitR2\": " << peak->gaussFitR2 ;
+            myfile << ",\n" << "\"peakRank\": " << peak->peakRank ;
+            myfile << ",\n" << "\"peakWidth\": " << peak->width ;
+        }
+        else {
+            myfile << "{\n";
+            myfile << "\"sampleName\": " << _sanitizeJSONstring((*it)->sampleName) ;
+            myfile << ",\n" << "\"peakMz\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"medianMz\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"baseMz\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"mzmin\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"mzmax\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"rt\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"rtmin\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"rtmax\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"quality\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"peakIntensity\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"peakBaseLineLevel\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"peakArea\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"peakSplineArea\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"peakAreaTop\": " << "\"NA\"";
+            myfile << ",\n" << "\"peakAreaNotCorrected\": " << "\"NA\"";
+            myfile << ",\n" << "\"peakAreaTopNotCorrected\": " << "\"NA\"";
+            myfile << ",\n" << "\"noNoiseObs\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"signalBaselineRatio\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"fromBlankSample\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"peakAreaFractional\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"symmetry\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"noNoiseFraction\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"groupOverlap\": " << "\"NA\"";
+            myfile << ",\n" << "\"groupOverlapFrac\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"gaussFitR2\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"peakRank\": " << "\"NA\"" ;
+            myfile << ",\n" << "\"peakWidth\": " << "\"NA\"" ;
+        }
+        
+        _writeEIC(grp,myfile,*it);
+    }
+    
+    myfile << "\n]" ; //peaks
+    
+    myfile << "}" ; //group
+}
+
+
+void JSONReports::_writeEIC(PeakGroup& grp, ofstream& myfile, mzSample* it){

```

Comment:
Name pointer to an `mzSample` as `sample` instead of `it`, which makes it look like an iterator.